### DISCUSSION TITLE
Firefox extension content script not installed after client-side page change

### DISCUSSION THREAD

**Discussion by t1v1** [vercel/next.js#34232] [Help]
### Run `next info` (available from version 12.0.8 and up)

```
/bin/sh: 1: pnpm: not found

    Operating System:
      Platform: linux
      Arch: x64
      Version: #202112220937~1640185481~20.04~b3a2c21-Ubuntu SMP Mon Jan 3 17:0
    Binaries:
      Node: 16.13.1
      npm: 8.1.2
      Yarn: 3.1.0
      pnpm: N/A
    Relevant packages:
      next: 12.0.9
      react: 17.0.2
      react-dom: 17.0.2
```

### What version of Next.js are you using?

12.0.9

### What version of Node.js are you using?

16.13.1

### What browser are you using?

Firefox

### What operating system are you using?

Pop_OS! 20.04 (Ubuntu 20.04)

### How are you deploying your application?

Vercel

### Describe the Bug

My [app](https://github.com/t1v1/thavma) includes a Firefox extension and communicates with said extension (to ensure that it's installed) via the `postMessage` API to a `content_script` installed on `thavma.io/assessments` and `localhost/assessments`:

```json
// ext/manifest.json
  "content_scripts": [
    {
      "matches": [
        "http://localhost/assessments",
        "http://localhost/assessments?*",
        "https://thavma.io/assessments",
        "https://thavma.io/assessments?*"
      ],
      "js": ["thavma.js"],
      "all_frames": true
    }
  ],
```

Based on that `manifest.json`, Firefox loads the `thavma.js` extension script whenever the URL is `/assessments`. This works perfectly fine when one types in the entire URL (i.e. `localhost:3000/assessments` or `thavma.io/assessments`) but when the user visits the home page (i.e. `localhost:3000/`) and then uses the navigation header to perform a client-side page change to `/assessments`, the `content_script` is not installed and thus isn't detected by my app.

### Expected Behavior

The way that Next.js performs client-side page transitions should notify Firefox (and Chromium-based browsers too, of course) to install any registered extension scripts to that URL.

### To Reproduce

You should be able to pretty easily reproduce with the original [`thavma`](https://github.com/t1v1/thavma) repository:
1. Clone the repository: `git clone https://github.com/t1v1/thavma`
2. Install dependencies: `cd thavma && yarn install`
3. Start the Next.js development server: `yarn dev`
4. Open Firefox, go to `about:debugging`, click "This Firefox", click "Load Temporary Add-on...", and select the `thavma/ext/manifest.json` file.
5. Open up the console and go to `localhost:3000/assessments` (if you hit a paywall, use the Stripe test card `4242 4242 4242 4242` and any CVC, postal code, etc).
6. See the logs from the `ext/thavma.js` script:

![extension-logs](https://user-images.githubusercontent.com/93789775/153329219-6a3222bb-1b14-4304-8e33-6079db8625ed.png)

7. Now, go to `localhost:3000/` and click "assessments" in the header to perform a client-side page change to `/assessments`.
8. Open up the console and notice that there are no logs from the `ext/thavma.js` script... it wasn't loaded by Firefox as expected.

If you're somehow unable to reproduce using those steps, or you really want something simpler, I can probably take a couple hours to create a smaller repro for you. I really think this bug shouldn't be overlooked in a framework as polished as Next.

**Comment by t1v1** [comment]
Temporary fix is to simply install that `content_script` on every single page within my application. This is far from ideal.

**Reply by balazsorban44** [reply]
Hi, forgive my lack of experience with Firefox extensions, but when following your reproduction steps  and visiting `http://localhost:3000`, I do see the extension is installed in the log:

![image](https://user-images.githubusercontent.com/18369201/153625225-5b7803d6-0cdc-49cd-ab60-a020f2dfd27c.png)

Am I missing something? If you could put together a more simple reproduction, that would probably help a lot in excluding some uncertainty about where your problem is coming from.
